cli/main.ml

```diff
@@ -160,22 +160,41 @@ module Impl = struct
     let filename = require "filename" filename in
     let t =
       let sock = Lwt_unix.socket Unix.PF_INET Unix.SOCK_STREAM 0 in
-      let sockaddr = Lwt_unix.ADDR_INET(Unix.inet_addr_any, port) in
-      Lwt_unix.bind sock sockaddr;
-      Lwt_unix.listen sock 5;
-      let rec loop () =
-        Lwt_unix.accept sock
-        >>= fun (fd, _) ->
-        (* Background thread per connection *)
-        let _ =
-          let channel = Nbd_lwt_unix.of_fd fd in
-          Server.connect channel ()
-          >>= fun (name, t) ->
-          Block.connect filename
-          >>= fun b ->
-          Server.serve t (module Block) b in
-        loop () in
-      loop () in
+      Lwt.finalize
+        (fun () ->
+           let sockaddr = Lwt_unix.ADDR_INET(Unix.inet_addr_any, port) in
+           Lwt_unix.bind sock sockaddr;
+           Lwt_unix.listen sock 5;
+           let rec loop () =
+             Lwt_unix.accept sock
+             >>= fun (fd, _) ->
+             (* Background thread per connection *)
+             let _ =
+               Lwt.try_bind
```

Comment:
Can we extract both try_binds in separate functions? The code is getting lesser and lesser readable